rcap/http_rulefilter_test.go

```diff
@@ -0,0 +1,222 @@
+package rcap
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestDefaultRules(t *testing.T) {
+	rules := defaultHTTPRules()
+
+	t.Run("http allowed", func(t *testing.T) {
+		req, _ := http.NewRequest(http.MethodGet, "http://example.com", nil)
+
+		if err := rules.requestIsAllowed(req); err != nil {
+			t.Error("error occurred, should not have")
+		}
+	})
+
+	t.Run("https allowed", func(t *testing.T) {
+		req, _ := http.NewRequest(http.MethodGet, "https://example.com", nil)
+
+		if err := rules.requestIsAllowed(req); err != nil {
+			t.Error("error occurred, should not have")
+		}
+	})
+
+	t.Run("IP allowed", func(t *testing.T) {
+		req, _ := http.NewRequest(http.MethodGet, "http://10.11.12.13", nil)
+
+		if err := rules.requestIsAllowed(req); err != nil {
+			t.Error("error occurred, should not have")
+		}
+	})
+}
+
+func TestAllowedDomains(t *testing.T) {
+	rules := defaultHTTPRules()
+	rules.AllowedDomains = []string{"example.com", "another.com", "*.hello.com", "tomorrow.*", "10.*.12.13"}
```

Comment:
```suggestion
	rules.AllowedDomains = []string{"example.com", "another.com", "*.hello.com", "tomorrow.*", "10.*.12.13", "example.com:8080"}
	
	t.Run("example.com:8080 allowed", func(t *testing.T) {
		req, _ := http.NewRequest(http.MethodGet, "http://example.com:8080", nil)

		if err := rules.requestIsAllowed(req); err != nil {
			t.Error("error occurred, should not have")
		}
	})

	t.Run("example.com:8081 disallowed", func(t *testing.T) {
		req, _ := http.NewRequest(http.MethodGet, "http://example.com:8081", nil)

		if err := rules.requestIsAllowed(req); err == nil {
			t.Error("error did not occur, should have")
		}
	})

```

We can just treat them as different domains altogether. These tests pass.